packages/wallets/trezor/src/adapter.ts

```diff
@@ -0,0 +1,169 @@
+    supportedTransactionVersions: ReadonlySet<TransactionVersion> = new Set(['legacy', 0]);
+
+    private _derivationPath: string;
+    private _connectUrl?: string;
+    private _connecting: boolean;
+    private _publicKey: PublicKey | null;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' ||
+        typeof document === 'undefined' ||
+        typeof navigator === 'undefined' ||
+        !navigator.hid
+            ? WalletReadyState.Unsupported
+            : WalletReadyState.Loadable;
+
+    constructor(config: TrezorWalletAdapterConfig = {}) {
+        super();
+        this._derivationPath = config.derivationPath || `m/44'/501'/0'/0'`;
+        this._connectUrl = config.connectUrl;
+        this._connecting = false;
+        this._publicKey = null;
+    }
+
+    get publicKey() {
+        return this._publicKey;
+    }
+
+    get connecting() {
+        return this._connecting;
+    }
+
+    get readyState() {
+        return this._readyState;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            if (this._readyState !== WalletReadyState.Loadable) throw new WalletNotReadyError();
+
+            this._connecting = true;
+
+            const connectUrl = this._connectUrl && this._connectUrl + (this._connectUrl.endsWith('/') ? '' : '/');
+
+            await TrezorConnect.init({
+                manifest: {
+                    email: 'maintainers@solana.foundation',
+                    appUrl: 'https://github.com/solana-labs/wallet-adapter',
+                },
+                lazyLoad: true,
+                ...(this._connectUrl
+                    ? {
+                          connectSrc: connectUrl,
+                          iframeSrc: connectUrl,
+                      }
+                    : {}),
+            });
+
+            let result;
+            try {
+                result = await TrezorConnect.solanaGetPublicKey({
+                    path: this._derivationPath,
+                });
+            } catch (error: any) {
+                throw new WalletPublicKeyError(error?.message, error);
+            }
+
+            if (!result.success) {
+                throw new WalletPublicKeyError(result.payload?.error, result.payload);
+            }
+
+            const publicKey = result.payload.publicKey;
+
+            TrezorConnect.on(DEVICE_EVENT, (event: any) => {
+                if (event.type === DEVICE.DISCONNECT) {
+                    this._disconnected();
+                }
+            });
+
+            this._publicKey = new PublicKey(Buffer.from(publicKey, 'hex'));
```

Comment:
Can you get your API to return a `Uint8Array` here? Solana public keys never use hex strings, and I'd prefer not to add the Buffer dependency here, we try to avoid it wherever possible. It's already been eliminated from the new version of web3.js we're moving to.

Reply:
> Can you get your API to return a Uint8Array here?

Sadly I don't think so. TrezorConnect accepts and returns bytes as hex for other blockchains as well. Also the Connect implementation has already been approved so I'd rather not introduce changes at this point - since we're very close to the release.

For now I'd suggest no action since I see the Solana SDK still returns a `Buffer` for `transaction.message.serialize()` and `transaction.serializeMessage()` and `transaction.addSignature` takes a `Buffer` as parameter.

What version of web3 will you be moving to? 2.0.0?

When Solana SDK no longer works with Buffer, I'd suggest we introduce local utils for hex <-> Uint8Array conversion and that would be all that is required.

What do you think?

Reply:
Okay, I think that's fine for now. I just don't want to end up importing a second copy of `Buffer`. I'll take a look at how this is getting built.

Reply:
(I resolved the other two comments I made, I'll leave this one pending for now, but I don't think there's any action you need to take on it.)